### DISCUSSION TITLE
Advantages/Disadvantages of Open and Closed Source Software

### DISCUSSION THREAD

**Discussion by jsxu315** [albertkun/21S-ASIAAM-191A#118] [Past Thinking Caps]
If there’s a problem with a closed-source tool, only the company can fix it; how fast it’ll be fixed and the general reliability of closed source software varies from company to company. Closed source does mean that companies have a financial motivation to produce good products, but companies only have to do as much as they need to to turn a profit, which doesn’t necessarily translate to a good experience for consumers. 

Proprietary software is also at the whims of the company that controls it, especially if there aren’t many major competitors. For example, if people want to use Adobe’s products, the options are to pay around $20/month for one app (expensive if you’re not part of a company/institution that gives you a license), or to pirate it (which can be inconvenient to update and risky if you’re new to torrenting). 

Open-source projects also make services accessible to people who can’t afford expensive software; Open Office, GIMP, Firefox, Git, etc., offer free alternatives to paid products (Microsoft Office suite, Photoshop). Closed source software is unlikely to be made or continue to be supported unless it’s commercially viable. Some programs that are useful for niche academic purposes (as an example) but not the general population might have never existed if not for open source developers. 

The ethics of both open- and closed-source software can be murky, but there have been examples of companies selling their software for unethical purposes (Palantir taking contracts for ICE, Amazon Ring collaborating with police departments I believe). While closed-source does mean that a company can choose not to sell to certain groups, it also means that the company can hide who it does choose to sell to. The existence of open-source software in the public eye means that at least people know exactly how a piece of software can be used and what they can do to protect themselves.

**Comment by vcta99** [comment]
Agree with you on the idea that open source software is definitely more accessible and operates more like a public good. Although conventional economics will tell us that private entities/individuals are unlikely to provide public goods (low benefit to cost ratio), I wonder, what motivates people to create open source software? Is it truly the spirit of altruism?

**Comment by jason-muljadi** [comment]
Very well thought out points and definitely agree that the closed-source tools used by the companies are up to the whims and discretion of the companies themselves. Capitalism drives the greed of the companies and you may not know what they are up to. 

However, I feel that people can just as easily capitalize on open-source software as well. For example, Roblox, a game in which I grew up playing religiously, has a huge open-source network that has allowed players to profit from it. There is a reason why the IPO had a lot of backing, but with many people using the OSS and in many creative ways, I feel there is always a risk for a cybersecurity attack as well.

**Comment by epham36** [comment]
You bring up such good points, especially bringing up how closed source software developers are incentivized to keep their software updated -- but to an extent -- and for purposes of financial profit, over consumer satisfaction, per se. The responsiveness of corporate software and services to what their consumers seek out and would actually find benefit them most -- is not necessarily what is profitable -- unless there is financial incentive to provide customer support, satisfaction, effective UI/UX, and proper functionalities.
